internal.nix

```diff
@@ -303,6 +306,16 @@ rec {
         cleanArgs = builtins.removeAttrs args [ "src" "packageJson" "packageLockJson" "buildInputs" "nativeBuildInputs" "nodejs" "preBuild" "postBuild" "preInstallLinks" "githubSourceHashMap" ];
         lockfile = readLockfile packageLockJson;
 
+        pnameComputed =
+          if (pname != "") then makeValidDrvName pname
+          else if (builtins.hasAttr "name" lockfile && lockfile.name != "") then lockfile.name
+          else ""; # pname is required, throw later
+
+        versionComputed =
+          if (version != "") then version
+          else if (builtins.hasAttr "version" lockfile && lockfile.version != "") then lockfile.version
+          else "0.0.0"; # version is optional
```

Comment:
print info message when using the unknown version 0.0.0?
usually the version is known, its just not specified in package-lock.json

Reply:
I think a trace here would be a good idea since npmlock2nix is applying changes here 👍